pkg/trustedresources/verify.go

```diff
@@ -41,6 +41,30 @@ const (
 	SignatureAnnotation = "tekton.dev/signature"
 )
 
+const (
+	IgnoreWhenNoMatchPolicy = iota
```

Comment:
Suggest more consistency in naming, e.g.
```
NoMatchingPoliciesIgnore
NoMatchingPoliciesWarn
NoMatchingPoliciesError
VerificationWarning
VerificationFailed
VerificationPassed
```
Also I think we still need a mode for no matching policies + failure?

Reply:
no matching policies + failure belongs to VerificationError, all previous returned error cases go into this case since we don't need to have different types for them, there will be only 1 case when updating the condition.

Reply:
How about this? 
```
VerificationError
VerificationWarn
VerificationSuccess
VerificationSkipped
```
I realized that NoMatchingPoliciesWarn and WarnModeVerificationFail are actually the same case when updating the condition

Reply:
My 2c:
what result mode we need here depends on what the tekton controller will be doing with different verification result and what level of details we want to surface to the users. @Yongxuanzhang Can you please add some details to the PR description about that?